### PR TITLE
net: openthread: Add PSA implementation for PBDKF2 genkey

### PR DESCRIPTION
**Pull request by maciejbaczmanski**
Add implementation of opentherad pbkdf2 generate key using PSA functions.

Based on: https://github.com/nrfconnect/sdk-zephyr/pull/1403

### CODE REVIEW

**Comment by pdgendt** [comment]
Nit: typo in commit message `s/opentherad/openthread/`

**Review by rlubos — DISMISSED** [review]

**Review by pdgendt — DISMISSED** [review]

File: modules/openthread/platform/crypto_psa.c [hunk 1]
```diff
@@ -603,4 +603,66 @@ otError otPlatCryptoEcdsaGenerateAndImportKey(otCryptoKeyRef aKeyRef)
...
+	if (status != PSA_SUCCESS) {
+		goto out;
+	}
+
+out:
+	psa_reset_key_attributes(&attributes);
+	psa_key_derivation_abort(&operation);
+	psa_destroy_key(key_id);
+
+	OT_ASSERT(status == PSA_SUCCESS);
```
Comment (maciejbaczmanski) [inline modules/openthread/platform/crypto_psa.c:664]: Found while testing: 
```suggestion
	__ASSERT_NO_MSG(status == PSA_SUCCESS);
```

`OT_ASSERT` is used by openthread internally. We should swap that with Zephyr's asserts

**Comment by maciejbaczmanski** [comment]
Found a bug while testing, device asserts when using pskcref. Adding DNM label until issue is investigated

**Comment by maciejbaczmanski** [comment]
> Found a bug while testing, device asserts when using pskcref. Adding DNM label until issue is investigated

Assert was connected with missing Kconfig in NCS, so it was not this implementation's fault.

Tested pskc generation with test vector from Thread specification 8.4.1.2.2 , resulting pskc was correct

**Review by pdgendt — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by edmont — APPROVED** [review]

**Pull request merged**